src/api/dotnet/Context.cs

```diff
@@ -312,9 +312,18 @@ public EnumSort MkEnumSort(Symbol name, params Symbol[] enumNames)
         public EnumSort MkEnumSort(string name, params string[] enumNames)
         {
             Debug.Assert(enumNames != null);
-	    using var sname = MkSymbol(name);
-	    using var snames = MkSymbols(enumNames);
-            return new EnumSort(this, sname, snames);
+
+            using var symbol = MkSymbol(name);
+            var symbols = MkSymbols(enumNames);
+            try 
+            {
+                return new EnumSort(this, symbol, symbols);
+            }
+            finally
+            {
+                foreach (var symbol in symbols)
```

Comment:
These collections of `Z3Object` were a bit annoying in that we can't just call `using var symbols = MkSymbols(enumNames)` because that will only dispose of the collection, which I do not believe dispose of the elements within the collection. It seems the only way to do it is to desugar the `using` can add an explicit `finally` block where enumerate the collection and manually `Dispose` each item.